Add instructions to preview docs site locally to /website readme (#29085)
hashicorp/terraform: website/README.md

[the `terraform-website` repository](https://github.com/hashicorp/terraform-website), which brings all of the
different documentation sources together and contains the scripts for testing and building the site as
a whole.

## Previewing Changes

> You should preview all of your changes locally before creating a pull request. The build includes content from this repository and the `terraform-website` repository, allowing you to preview the entire Terraform documentation site.

Reviewer: ```suggestion
You should preview all of your changes locally before creating a pull request. The build includes content from this repository and the [`terraform-website`](https://github.com/hashicorp/terraform-website/) repository, allowing you to preview the entire Terraform documentation site. If `terraform-website` isn't in your `GOPATH`, the preview command will clone it to your machine.
```

**Set Up Local Environment**

1. [Install Docker](https://docs.docker.com/get-docker/).

> 2. Create a ~/go directory manually or by [installing Go](https://golang.org/doc/install).
> 3. Open terminal and set GOPATH as an environment variable:
> 
>     Bash: `export $GOPATH=~/go`(bash)
> 
>     Zsh: `echo -n 'export GOPATH=~/go' >> ~/.zshrc`
> 4. Restart your terminal or command line session.

Reviewer: ```suggestion
1. Create a `~/go` directory manually or by [installing Go](https://golang.org/doc/install).
1. Open terminal and set `GOPATH` as an environment variable:

    Bash: `export $GOPATH=~/go`(bash)

    Zsh: `echo -n 'export GOPATH=~/go' >> ~/.zshrc`
1. Restart your terminal or command line session.
```

**Launch Site Locally**

1. Navigate into your local `terraform` top-level directory and run `make website`.
3. Open `http://localhost:4567` in your web browser. While the preview is running, you can edit pages and Middleman will automatically rebuild them.

> 4. When you're done with the preview, press ctrl-C in your terminal to stop the server.

Reviewer: ```suggestion
4. When you're done with the preview, press `ctrl-C` in your terminal to stop the server.
```
